pinot-controller/src/main/java/org/apache/pinot/controller/api/upload/ZKOperator.java

```diff
@@ -65,7 +65,14 @@ public void completeSegmentOperations(String tableNameWithType, SegmentMetadata
     String segmentName = segmentMetadata.getName();
     ZNRecord segmentMetadataZNRecord =
         _pinotHelixResourceManager.getSegmentMetadataZnRecord(tableNameWithType, segmentName);
+    boolean refreshOnly =
+        Boolean.parseBoolean(headers.getHeaderString(FileUploadDownloadClient.CustomHeaders.REFRESH_ONLY));
     if (segmentMetadataZNRecord == null) {
+      if (refreshOnly) {
+        throw new ControllerApplicationException(LOGGER,
+            "Cannot refresh non-existing segment, aborted uploading segment: " + segmentName + " of table: "
+                + tableNameWithType, Response.Status.GONE);
```

Comment:
If we decided to use a new response status, please update description in the resource level as well.

Reply:
Good point, added the api response code in api description.